test/metabase/api/advanced_computation_test.clj

```diff
@@ -63,6 +71,24 @@
           (is (= ["MS" nil "Doohickey" 78 291] (drop-last (nth rows 1000))))
           (is (= [nil nil nil 18760 69540] (drop-last (last rows)))))))))
 
+(deftest pivot-filter-dataset-test
+  (mt/dataset sample-dataset
+    (testing "POST /api/advanced_computation/pivot/dataset"
+       (testing "Run a pivot table"
+         (let [result (mt/user-http-request :rasta :post 202 "advanced_computation/pivot/dataset" (filters-query))
+               rows   (mt/rows result)]
+           (is (= 140 (:row_count result)))
+           (is (= "completed" (:status result)))
+           (is (= 4 (count (get-in result [:data :cols]))))
+           (is (= 140 (count rows)))
+
+           ;; spot checking rows, but leaving off the discriminator on the end
+           (is (= ["AK" "Google" 119] (drop-last (first rows))))
+           (is (= ["AK" "Organic" 89] (drop-last (second rows))))
+           (is (= [nil "Google" 3798] (drop-last (nth rows 90))))
+           (is (= [nil "Organic" 3764] (drop-last (nth rows 91))))
+           (is (= [nil nil 7562] (drop-last (last rows)))))))))
+
 (deftest pivot-card-test
```

Comment:
Maybe it's worth adding a test of parameters in here.

Reply:
yep that's my plan! I just didn't get to it yet.